test/unit/typedQueryBuilderTests.ts

```diff
@@ -1405,6 +1406,61 @@ describe('TypedKnexQueryBuilder', () => {
         );
     });
 
+    it('should get correct column for nested queries', (done) => {
+        const typedKnex = new TypedKnex(knex({ client: 'postgresql' }));
+        let nestedStatusColumnName;
+        let nestedUserCategoryYear;
+        const query = typedKnex.query(User)
+            .whereExists(User, (subQuery) => {
+                nestedStatusColumnName = subQuery.getColumnAlias('status');
+                nestedUserCategoryYear = subQuery.getColumnAlias('category.year');
```

Comment:
Okay I tried again. Part of it works, but part of it doesn't. I logged the `subQueryPrefix` and the alias result from the `getColumnAlias` function for this line and the line above:

> subQueryPrefix is:  subquery0$
> And columnAlias is:  "subquery0$users"."weirdDatabaseName"

> subQueryPrefix is:  subquery0$
> And columnAlias is:  "category"."year"


So it seems that if you try to access a column of the current table (e.g. `'status'`), the subquery is added correctly. But if you try to access a column of a foreign key linked table (e.g. `'category.year'`), you don't get the subquery added.

Can you get that to work?

Reply:
Done!